Rel 1158490 doc improvement

Chaitan2681 opened the pull request:
EW Documentation : Single PR for all Docs as Continuous Improvement Story

Jira Ticket : https://jira.kcura.com/browse/REL-1158490

Test Evidence for rabbitmq_integration.md file : 
[rabbitmq_integration.md file Test evidence.docx](https://github.com/user-attachments/files/22042381/rabbitmq_integration.md.file.Test.evidence.docx)

Test Evidence for elasticsearch_pre_installation_overview.md file
[elasticsearch_pre_installation_overview.docx](https://github.com/user-attachments/files/22042779/elasticsearch_pre_installation_overview.docx)

Test Evidence for  elasticsearch_setup_development.md file
[elasticsearch_setup_development.docx](https://github.com/user-attachments/files/22048353/elasticsearch_setup_development.docx)

Test Evidence for monitoring-agent-and-otel-collector.md file
[monitoring-agent-and-otel-collector-md-test-evidence.docx](https://github.com/user-attachments/files/22043102/monitoring-agent-and-otel-collector-md-test-evidence.docx)

Test Evidence for relativity_server_cli_setup.md file:
[relativity_server_cli_setup.docx](https://github.com/user-attachments/files/22043720/relativity_server_cli_setup.docx)

Test Evidence for relativity-server-cli.md file
[relativity-server-cli-md-test-evidence.docx](https://github.com/user-attachments/files/22043861/relativity-server-cli-md-test-evidence.docx)

Test Evidence for docs/datagrid_audit_setup.md. file
[DataGrid Audit.docx](https://github.com/user-attachments/files/22044960/DataGrid.Audit.docx)


Test Evidence for install_environment_watch_monitoring_agents.md file
[install_environment_watch_monitoring_agents.docx](https://github.com/user-attachments/files/22043936/install_environment_watch_monitoring_agents.docx)

Test Evidence to verify the 7 steps and navigation to the next steps : 
[Test Evidence to verify 7 steps and navigation .docx](https://github.com/user-attachments/files/22046640/Test.Evidence.to.verify.7.steps.and.navigation.docx)

Test Evidence for Troubleshooting(elasticsearch.md, kibana.md and apm-server.md)
[Evidence for Troubleshooting(ES,Kibana,APM-Server).docx](https://github.com/user-attachments/files/22047012/Evidence.for.Troubleshooting.ES.Kibana.APM-Server.docx)

deepankarrelativity002 committed:
REL-1158490 resolving minor conflicts- changing terminology, removing pre-requisite, updating Readme.md file

deepankarrelativity002 committed:
Merge branch 'main' into REL-1158490-Doc_improvement

deepankarrelativity002 committed:
REL-1158490 updating elasticsearch_pre_installation_overview and ew installer md file

deepankarrelativity002 committed:
REL-1158490 updating elasticsearch_pre_installation_overview and ew installer md file

deepankarrelativity002 committed:
REL-1158490 updating elasticsearch_pre_installation_overview and ew installer md file

Chaitan2681 committed:
REL-1158490: Updated readme.md file and Environment watch installation.md files

Chaitan2681 committed:
REL-1158490: Updated the EW documentation

deepankarrelativity002 committed:
REL-1158490 updating elasticsearch_pre_installation_overview and ew installer md file

deepankarrelativity002 committed:
REL-1158490 updating readme.md file

Chaitan2681 committed:
REL-1158490: updating readme.md file

Divyavenkataraju committed:
REL-1158490_Updated  feedback comments

Chaitan2681 committed:
REL-1158490: Updated the feedback comments

Divyavenkataraju committed:
REL-1158490- updated  docs. based on review comments: moved System requirements to Readme.md, updated existing Development Tier title. Added consistent next step title

shalinisundarpraksh committed:
REL-1158490: Updated Rabbimq management plugin

Chaitan2681 committed:
REL-1158490: Updated the EW Documentation

Divyavenkataraju committed:
REL-1158490-Seperated CLI and Datagrid Audit Steps.Updated screenshots.

Divyavenkataraju committed:
REL-1158490: 1. Updated Steps in CLI and Datagrid files
                        2. As per review comments, removed old troubleshooting guide, added those steps in seperate sections under troubleshooting folder.
                        3. Updated links pointing to old troubleshooting guide
                        4. updated broken link for rabbitmq

Divyavenkataraju committed:
REL-1158490: As per the latest comments received by scott and karuna. 1. Renamed Readme.md containing postinstall steps to post-install-verfication.md
2. Moved  user creation from Alerts installation file to last step : Create kibana users
3. Updated next step link under rabbitmq file

Divyavenkataraju committed:
REL-1158490: Updated broken links

Divyavenkataraju commented on docs/datagrid_audit_setup.md:
```diff
@@ -0,0 +1,118 @@
+
+2. [Data Grid Audit only] Install the mapper-size plugin on all nodes in your Elasticsearch cluster (instructions available [here](https://www.elastic.co/guide/en/elasticsearch/plugins/current/mapper-size.html)). You also must restart the Elasticsearch service after installing the plugin.<br/>
+3. [Data Grid Audit only] Before upgrading to Elasticsearch 8.17, the ESIndexCreationSetting may need to be updated. For further details, please refer to the [Instance settings' descriptions - Server2024](https://help.relativity.com/Server2024/Content/System_Guides/Instance_Setting_Guide/Instance_setting_descriptions.htm#ESIndexCreationSettings).<br/>
+4. At least the minimum Relativity major version and patch specified in the Environment Watch bundle you intend to deploy is installed on all servers in the environment. See the [release bundle](https://github.com/relativitydev/server-bundle-release/releases) requirements for the minimum version required.<br/>
```
Updated the link to README.md

Divyavenkataraju commented on docs/environment_watch_RabbitMQ_Management_Plugin.md:
we are following small cases, file name updated

Divyavenkataraju commented on docs/environment_watch_installation.md:
```diff
@@ -8,28 +8,41 @@ Environment Watch and Data Grid Audit require installation and configuration of
-The Server bundle is generally released quarterly but hot fixes are provided for critical issues on an ad hoc basis.
 
-Environment Watch installation is comprised of the following four stages. **Stages 1 and 2 are also used to set up Elasticsearch and Kibana for using Data Grid Audit. Stages 3 and 4 are only relevant for Environment Watch.**
+The Server bundle is generally released quarterly, with hot fixes provided for critical issues as needed.
```
updated

Divyavenkataraju commented on docs/environment-watch/post-install-verification/monitoring-agents.md:
```diff
@@ -148,3 +147,7 @@ Ensure dashboard data is reflected in Discover.
+
+## Next Step
+
+[Click here for the next step](alerts.md)
```
yes

Divyavenkataraju committed:
REL-1158490: Updated files as per review comments

Divyavenkataraju committed:
REL-1158490-Renamed filename and navigation

KarunaDhawan commented on README.md:
```diff
@@ -1,25 +1,18 @@
 
-- **Environment Watch**: Environment Watch is a comprehensive monitoring solution built on the Elastic stack designed for Relativity Server customers, providing deep insights into their system’s health and performance. By leveraging OpenTelemetry, it collects and structures telemetry data (metrics, logs, and traces) within an Elasticsearch cluster hosted in the user’s environment, ensuring data s ...
-- **Data Grid Audit**: Audit is a Relativity application that you can use to monitor and run reports on audited user activity. You must have Elasticsearch configured to use the Audit application. This release bundle includes a setup package and documentation for configuring Elasticsearch for Data Grid Audit. Additional Audit setup instructions are available [here](https://help.relativity.com/Serv ...
+- **Environment Watch**: Monitoring solution for Relativity Server, built on the Elastic stack. Collects metrics, logs, and traces using OpenTelemetry, stores data in your own Elasticsearch cluster, and provides real-time alerts and dashboards. [Click here to learn more about Environment Watch](docs/environment_watch_product_overview.md)
```
Update the description to -> Environment Watch is a monitoring solution for Relativity Server built on the Elastic stack. Using OpenTelemetry, it captures metrics, logs, and traces into a customer-hosted Elasticsearch cluster, ensuring data security and control. Key features include real-time alerts, dashboards, log search in Kibana, and in-app notifications within Relativity to help admins quickly detect and resolve issues.

KarunaDhawan commented on README.md:
```diff
@@ -1,25 +1,18 @@
-- **Environment Watch**: Environment Watch is a comprehensive monitoring solution built on the Elastic stack designed for Relativity Server customers, providing deep insights into their system’s health and performance. By leveraging OpenTelemetry, it collects and structures telemetry data (metrics, logs, and traces) within an Elasticsearch cluster hosted in the user’s environment, ensuring data s ...
-- **Data Grid Audit**: Audit is a Relativity application that you can use to monitor and run reports on audited user activity. You must have Elasticsearch configured to use the Audit application. This release bundle includes a setup package and documentation for configuring Elasticsearch for Data Grid Audit. Additional Audit setup instructions are available [here](https://help.relativity.com/Serv ...
+- **Environment Watch**: Monitoring solution for Relativity Server, built on the Elastic stack. Collects metrics, logs, and traces using OpenTelemetry, stores data in your own Elasticsearch cluster, and provides real-time alerts and dashboards. [Click here to learn more about Environment Watch](docs/environment_watch_product_overview.md)
+- **Data Grid Audit**: Application for monitoring and reporting on audited user activity. Requires Elasticsearch. [Click here to learn more about Data Grid Audit and setup instructions](https://help.relativity.com/Server2024/Content/Relativity/Audit/Audit.htm#InstallingandconfiguringAudit)
```
Update the description to -> Audit is a Relativity application for monitoring and reporting on user activity, requiring Elasticsearch. This bundle provides the setup package and configuration docs for Data Grid Audit.

KarunaDhawan commented on README.md:
```diff
@@ -1,25 +1,18 @@
 
-A list of common installation issues and their resolutions are available at [Environment Watch troubleshooting guide](docs/environment_watch_troubleshooting.md).
\ No newline at end of file
+- [Environment Watch and Data Grid Audit Installation](docs/environment_watch_installation.md)
```
Can we add a line here? like in original .md 'For installation instructions, click here

KarunaDhawan replied:
@Divyavenkataraju Pls remove Product overview link under 'Getting Started'. IT shoudl only have 2 links 
Installation and Troubleshooting

Divyavenkataraju replied:
Removed Product Overview Link

Chaitan2681 committed:
REL-1158490: Updated EW_troubleshooting.md file

Rahiman-Nadaf committed:
REL-1158490: Updated SSL certficate troubleshooting steps

Divyavenkataraju committed:
REL-1158490-Updated Review Comments. Formatted elasticsearch.md file

KarunaDhawan commented on README.md:
```diff
@@ -1,25 +1,21 @@
-| Relativity.Alerts.24000.0.xxxx.rap  | Used to view alerts within Relativity Alerts.                                                                   | N/A                                                                                   |
+| Requirement                       | Version/Details                                           |
+| --------------------------------- | --------------------------------------------------------- |
+| Minimum Relativity Server version | Server 2024 Patch 2                                       |
```
Can we update this to "Relativity Server"

Divyavenkataraju replied:
Updated

KarunaDhawan commented on README.md:
```diff
@@ -1,25 +1,21 @@
+| Requirement                       | Version/Details                                           |
+| --------------------------------- | --------------------------------------------------------- |
+| Minimum Relativity Server version | Server 2024 Patch 2                                       |
+| ElasticStack version              | **8.17.3** (Required for both Environment Watch and Data Grid Audit) |
```
Update this to "Elastic Stack"

Divyavenkataraju replied:
Updated

Divyavenkataraju committed:
REL-1158490: Updated docs based on review comments shared by karuna

KarunaDhawan commented on docs/troubleshooting/elasticsearch.md:
```diff
@@ -7,19 +7,21 @@ This document provides troubleshooting guidance for common Elasticsearch issues
 
 ## Table of Contents
```
1. **Banner under step images**  
   - Add a banner directly under each step image to indicate whether the step applies to **Environment Watch only** or to **both Environment Watch and Data Grid Audit**.  
   - Use the following formats:  
     - **_This step is required for both Environment Watch and Data Grid Audit_**  
     - _**This step is required for Environment Watch**_  

2. **Pre-requisites**  
   - Do not repeat pre-requisite steps in every file.  
   - Remove the **"Before you start"** section, as these steps are already covered earlier.  

3. **Content alignment with public doc SOP**  
   - Step 1: Installing Elastic certificate on all servers should already be covered in pre-installation. Ensure it’s not duplicated here.  
   - Step 2: Current wording is unclear — *"Download the CLI from here, download the release bundle."* Please rewrite for clarity.  
   - Step 3: Maintain consistent terminology, e.g., always use **"Elevated Command Prompt"**.  

Overall, the documentation should align with our public documentation SOP. Please update accordingly.

KarunaDhawan commented on docs/create_kibana_users.md:
```diff
@@ -0,0 +1,27 @@
+
+### Congratulations!
+
+You have reached the end of the setup process. If you are still running into any issues, please refer to the  [Troubleshooting Guides](/docs/troubleshooting.md) for further assistance.
```
The link of Troubleshooting doesn't work. Let's ensure all the links are working and being directed to the correct location.

Divyavenkataraju replied:
Updated the troubleshooting doc link to environment_watch_troubleshooting.md

Divyavenkataraju commented on docs/environment_watch_install_other_integrations.md:
```diff
@@ -11,3 +11,7 @@ This guide provides information on integrating various Relativity components wit
+
+## Next Step
+
+[Click here for the next step](environment_watch_rabbitmq_management_plugin.md)
```
1. Created a new rabbitmq directory similar to analytics.
2. Moved the RabbitMQ content to a new rabbitmq_integration.md file.
3. Updated environment_watch_install_other_integrations.md to have a separate "RabbitMQ Integration" header and link to the new page.
4. Updated the "Next Step" link to point to the next logical step in the installation sequence.
5.Removed the old environment_watch_rabbitmq_management_plugin.md file.

Divyavenkataraju committed:
REL-1158490
1. Updated troubleshooting link pointing to old file.
2. Created a new rabbitmq directory.
3. Moved the RabbitMQ content to a new rabbitmq_integration.md file.
4. Updated environment_watch_install_other_integrations.md to have a separate "RabbitMQ Integration" header and link to the new page.
5. Updated the "Next Step" link to point to the next logical step in the installation sequence.
6.Removed the old environment_watch_rabbitmq_management_plugin.md file.

Divyavenkataraju commented on docs/environment_watch_installation.md:
```diff
@@ -8,28 +8,75 @@ Environment Watch and Data Grid Audit require installation and configuration of
-## Next 
-Proceed to stage 1: [Installing Elasticsearch, Kibana, and APM Server](elasticsearch_pre_installation_overview.md)
\ No newline at end of file
+<table>
```
Removed column from the table which was requested earlier

Divyavenkataraju committed:
REL-1158490- Updated Review comments shared by karuna: As requested removed the  columns EW and Datagrid check marks.

Divyavenkataraju committed:
REL-1158490: Updated Review comments

Chaitan2681 committed:
REL-1158490: Updated the feedbacks given by the SQE team

Chaitan2681 committed:
Merge branch 'REL-1158490-Doc_improvement' of https://github.com/relativitydev/server-bundle-release into REL-1158490-Doc_improvement

deepankarrelativity002 committed:
REL-1158490 Updated changes for Secret Store section in monitoring agent and otel collector.md file

deepankarrelativity002 committed:
Merge branch 'REL-1158490-Doc_improvement' of https://github.com/relativitydev/server-bundle-release into REL-1158490-Doc_improvement

deepankarrelativity002 committed:
REL-1158490 Updated the monitoring agent and otel collector.md file

Divyavenkataraju committed:
REL-1158490: Updated CLI and Monitoring agents.md files as per review comments

Divyavenkataraju committed:
Merge branch 'REL-1158490-Doc_improvement' of https://github.com/relativitydev/server-bundle-release into REL-1158490-Doc_improvement

Divyavenkataraju committed:
REL-1158490: Updated missing link

Divyavenkataraju committed:
REL-1158490: Updated pre-requisites for EW installer

deepankarrelativity002 committed:
REL-1158490 Updated the datagrid audit setup.md file

deepankarrelativity002 committed:
REL-1158490 Updated the datagrid audit setup.md file

Divyavenkataraju committed:
REL-1158490: Updated Review comments

Divyavenkataraju commented on docs/install_environment_watch_monitoring_agents.md:
```diff
@@ -7,13 +7,10 @@ The Relativity Environment Watch installer package contains a monitoring agent (
-- SQL Primary and Distributed access
-- Relativity access
-- The minimum Relativity major version and patch is installed on all servers in the environment. See the [release bundle](https://github.com/relativitydev/server-bundle-release/releases) requirements for the minimum version required.
+- The Environment Watch monitoring agent must be installed on a supported Windows Server OS including Windows Server 2022, Windows Server 2019, Windows Server 2016. Please see [here](https://help.relativity.com/Server2024/Content/Installing_and_Upgrading/System_requirements/Compatibility_matrix.htm#Relativitysystemrequirementsmatrix) for information on Windows Server compatibility for Relativity  ...
```
Added pre-requisites. Aligned with public doc SOP

Divyavenkataraju commented on docs/relativity_server_cli_setup.md:
Updated Pre-requisites. Replaced screenshots with Code blocks. 
@KarunaDhawan , Since we have segregated EW and Datagrid Setup. Not getting how to add the banners

Divyavenkataraju committed:
REL-1158490: Updated Review comments shared by deepankar

Divyavenkataraju committed:
REL-1158490: Updated Documents based on review comments received on 28th and 29th

Divyavenkataraju committed:
REL-1158490: Updated banner having issues

Chaitan2681 committed:
REL-1158490: Updated the rabbitmq_integration.md file

Chaitan2681 committed:
REL-1158490: Updated the rabbitmq_integration.md file

Chaitan2681 committed:
REL-1158490: Updated the rabbitmq_integration.md file

Chaitan2681 committed:
REL-1158490: Updated the rabbitmq_integration.md file

dinesh1010101 commented:
Attached evidence for monitoring-agent-and-otel-collector.md file in PR description and also added couple of PR feedback on this file.

Chaitan2681 committed:
REL-1158490: Updated the rabbitmq_integration.md file

Divyavenkataraju committed:
REL-1158490: Updated CLI documentation

Divyavenkataraju committed:
Merge branch 'REL-1158490-Doc_improvement' of https://github.com/relativitydev/server-bundle-release into REL-1158490-Doc_improvement

dinesh1010101 commented on docs/troubleshooting/relativity-server-cli.md:
1. Elastic APM Integration Package looks like setup page, not troubleshooting steps like checking "Elastic APM" exist in "Installed Integration".
<img width="868" height="516" alt="image" src="https://github.com/user-attachments/assets/47a6d359-37ff-4276-ace8-3ed6d681fef8" />

Steps needs to updated more like troubleshooting way.
<img width="940" height="478" alt="image" src="https://github.com/user-attachments/assets/da917c71-0cfc-4edf-93a6-0794935d1b1c" />

dinesh1010101 commented:
Attached evidence for relativity-server-cli.md file in PR description and also added a PR feedback on this file.
https://github.com/relativitydev/server-bundle-release/pull/35#discussion_r2309845480

deepankarrelativity002 committed:
REL-1158490 Updated troubleshooting(ES,Kibana,APM)files

Divyavenkataraju committed:
REL-1158490: Updated Datagrid document. Removed reduntant data

Divyavenkataraju committed:
REL-1158490:Updated minor changes for CLI and Datagrid

Divyavenkataraju committed:
REL-1158490: Updated Document issues

Chaitan2681 committed:
REL-1158490: Updated troubleshooting files as per comments

deepankarrelativity002 committed:
REL-1158490 Updated Post Installation files

deepankarrelativity002 committed:
REL-1158490 Updated Post Installation files

Chaitan2681 commented on docs/troubleshooting/monitoring-agent-and-otel-collector.md:
Updated the changes as per review comments

dinesh1010101 commented:
@Chaitan2681 Could you verify point no 3 and 6 changes pushed? And also, below output is wrong I think some format issue.
<img width="545" height="209" alt="image" src="https://github.com/user-attachments/assets/757e4784-d3c7-4af2-94b8-f18e9c6766d1" />

Also could you verify this path?
<img width="346" height="74" alt="image" src="https://github.com/user-attachments/assets/94baf440-5ccf-4310-8979-9a4ad910a06c" />

deepankarrelativity002 committed:
REL-1158490 Updated troubleshooting(ES,Kibana,APM)files

Chaitan2681 committed:
REL-1158490: Updated the links and format

Chaitan2681 committed:
Merge branch 'REL-1158490-Doc_improvement' of https://github.com/relativitydev/server-bundle-release into REL-1158490-Doc_improvement

shalinisundarpraksh committed:
REL-1158490: updated next step for last alert overview post install verification

shalinisundarpraksh committed:
REL-1158490: Updated next link

KarunaDhawan reviewed (approved):
Approving this PR with the note that the README file must be updated immediately to include port information and clarify the one-time setup and upgrade line. I understand upgrade steps aren’t available yet, but all feedback—including upgrade instructions—needs to be addressed by Tuesday. Thanks! cc @scott-parillo

Chaitan2681 committed:
REL-1158490: Updated all the feedbacks from Scott and Karuna

Chaitan2681 committed:
Merge branch 'REL-1158490-Doc_improvement' of https://github.com/relativitydev/server-bundle-release into REL-1158490-Doc_improvement

Chaitan2681 committed:
Remove port requirement note from README

Removed important note about ensuring open ports for Elastic stack.

Chaitan2681 commented:
@KarunaDhawan, 

- We have updated Readme.md file by removing upgrade text and updated port diagram link.
- Removed all the backward and next links for post installation and troubleshooting guidelines.
- Renamed production setup title.
- Updated secret store title instead of secret server.

KarunaDhawan commented on README.md:
```diff
@@ -1,25 +1,25 @@
+| Requirement                       | Version/Details                                           |
+| --------------------------------- | --------------------------------------------------------- |
+| Relativity Server                 | Server 2024 Patch 2                                       |
+| ElasticStack                      | 8.17.3 is required for Environment Watch.<br/>7.17.x can be used but for Data Grid Audit only. |
```
Just a quick note: Elastic Stack should be written as two separate words. Let’s make sure we’re using the correct terminology consistently throughout the documentation.

Also, please add another row to the System Requirements table for Network Ports with the following details:

Network Ports | Ensure required ports are open for Elastic Stack communication. Refer to the port diagram for full network configuration details.

Chaitan2681 committed:
Fix formatting and add network ports information

Updated ElasticStack to Elastic Stack and added network ports requirement.

Chaitan2681 committed:
Fix link path in README for port diagram

Chaitan2681 committed:
Update link to port diagram in README.md
